src/Umbraco.Cms.ManagementApi/Controllers/NewInstallController.cs

```diff
@@ -0,0 +1,131 @@
+using Umbraco.Cms.Core.Hosting;
+using Umbraco.Cms.Core.Install.Models;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.Infrastructure.Install;
+using Umbraco.Cms.Infrastructure.Migrations.Install;
+using Umbraco.Cms.ManagementApi.Filters;
+using Umbraco.Cms.ManagementApi.ViewModels.Installer;
+using Umbraco.Extensions;
+using Umbraco.New.Cms.Core.Factories;
+using Umbraco.New.Cms.Core.Models.Installer;
+using Umbraco.New.Cms.Core.Services.Installer;
+using Umbraco.New.Cms.Web.Common.Routing;
+
+namespace Umbraco.Cms.ManagementApi.Controllers;
+
+[ApiController]
+[ApiVersion("1.0")]
+[BackOfficeRoute("api/v{version:apiVersion}/install")]
+[RequireRuntimeLevel(RuntimeLevel.Install)]
+public class NewInstallController : Controller
+{
+    private readonly IUmbracoMapper _mapper;
+    private readonly IInstallSettingsFactory _installSettingsFactory;
+    private readonly IInstallService _installService;
+    private readonly GlobalSettings _globalSettings;
+    private readonly IHostingEnvironment _hostingEnvironment;
+    private readonly InstallHelper _installHelper;
+    private readonly DatabaseBuilder _databaseBuilder;
+
+    public NewInstallController(
+        IUmbracoMapper mapper,
+        IInstallSettingsFactory installSettingsFactory,
+        IInstallService installService,
+        IOptions<GlobalSettings> globalSettings,
+        IHostingEnvironment hostingEnvironment,
+        InstallHelper installHelper,
+        DatabaseBuilder databaseBuilder)
+    {
+        _mapper = mapper;
+        _installSettingsFactory = installSettingsFactory;
+        _installService = installService;
+        _globalSettings = globalSettings.Value;
+        _hostingEnvironment = hostingEnvironment;
+        _installHelper = installHelper;
+        _databaseBuilder = databaseBuilder;
+    }
+
+    [HttpGet("settings")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status428PreconditionRequired)]
+    [ProducesResponseType(typeof(InstallSettingsViewModel), StatusCodes.Status200OK)]
+    public async Task<ActionResult<InstallSettingsViewModel>> Settings()
+    {
+        // Register that the install has started
+        await _installHelper.SetInstallStatusAsync(false, string.Empty);
+
+        InstallSettingsModel installSettings = _installSettingsFactory.GetInstallSettings();
+        InstallSettingsViewModel viewModel = _mapper.Map<InstallSettingsViewModel>(installSettings)!;
+
+        return viewModel;
+    }
+
+    [HttpPost("setup")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status428PreconditionRequired)]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> Setup(InstallViewModel installData)
+    {
+        if (ModelState.IsValid is false)
```

Comment:
It would be nice to perform ModelState validation as an `ActionFilterAttribute` instead... makes for less repetitive coding. See https://docs.microsoft.com/en-us/aspnet/web-api/overview/formats-and-model-binding/model-validation-in-aspnet-web-api#handling-validation-errors for inspiration.

Reply:
Very good point, I can do you one better even, API controller does the check by default, so I've just removed the check entirely 😄